### ISSUE TITLE
False positive error from object-shorthand 

### ISSUE THREAD

**Issue by KeriWarr** [eslint/eslint#6060]
**What version of ESLint are you using?**
2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
        "jsx": true
    }
  },
  "extends": "airbnb",
  "rules": {
    "react/prefer-stateless-function": 0,
    "prefer-template": 0
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
Here is a simplified version:
```javascript
const a = 1;

export const b = {
  [a]: a,
};
```

**What did you expect to happen?**
no linting error

I want `b` to be equal to: `{ '1': 1 }` and NOT `{ a: 1 }`

**What actually happened? Please include the actual, raw output from ESLint.**
```
.../test.js
  4:3  error  Expected property shorthand  object-shorthand
```

**Comment by kaicataldo** [comment]
Thanks for the report. This is a duplicate of https://github.com/eslint/eslint/issues/6015 and should be fixed in the next release.
